common/djangoapps/enrollment/views.py

```diff
@@ -555,12 +556,16 @@ def post(self, request):
 
         try:
             # Lookup the user, instead of using request.user, since request.user may not match the username POSTed.
-            user = User.objects.get(username=username)
+            if not email:
+                user = User.objects.get(username=username)
+            else:
+                user = User.objects.get(email=email)
+                username = user.username
         except ObjectDoesNotExist:
             return Response(
                 status=status.HTTP_406_NOT_ACCEPTABLE,
                 data={
-                    'message': u'The user {} does not exist.'.format(username)
+                    'message': u'The user {} does not exist.'.format(email if email else username)
```

Comment:
how about `email or username`

Reply:
fixed in https://github.com/raccoongang/edx-platform/pull/2069